pkg/funky/transformer.go

```diff
@@ -0,0 +1,102 @@
+package funky
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"os"
+)
+
+// RequestTransformer general interface for transforming http requests into Dispatch requests
+type RequestTransformer interface {
+	Transform(req *http.Request) (*Request, error)
+}
+
+// DefaultRequestTransformer transforms a request into a format suitable for Dispatch language servers
+type DefaultRequestTransformer struct {
+	timeout int
+	secrets []string
+	rw      HTTPReaderWriter
+}
+
+// NewDefaultRequestTransformer constructs a new DefaultRequestTransformer
+func NewDefaultRequestTransformer(timeout int, secrets []string, rw HTTPReaderWriter) RequestTransformer {
+	return &DefaultRequestTransformer{
+		timeout: timeout,
+		secrets: secrets,
+		rw:      rw,
+	}
+}
+
+// Transform transforms an http request into a Dispatch request.
+func (r *DefaultRequestTransformer) Transform(req *http.Request) (*Request, error) {
+	var body Request
+
+	// populate function timeout
+	ctx := map[string]interface{}{
+		"timeout": r.timeout,
+	}
+
+	// populate secrets
+	if len(r.secrets) != 0 {
```

Comment:
Is there a way this could be broken out in a more generic or pluggable way.  I think with the current dispatch, we have different middlewares that apply content to the context, can we use a similar approach?